sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/IssueLocationCollector.cs

```diff
@@ -208,29 +291,50 @@ private IEnumerable<string> GetIssueIds(Match match)
 
         private int GetOffset(Match match)
         {
-            var sign = match.Groups["sign"];
             var offset = match.Groups["offset"];
-            if (!sign.Success && !offset.Success)
+            return offset.Success ? int.Parse(offset.Value) : 0;
+        }
+
+        private static void EnsureNoDuplicatedPrimaryIds(IList<IIssueLocation> mergedLocations)
+        {
+            var duplicateLocationsIds = mergedLocations
+                            .Where(location => location.IsPrimary)
+                            .GroupBy(x => x.IssueId)
+                            .Where(x => x.Key != null)
+                            .FirstOrDefault(group => group.Count() > 1);
+            if (duplicateLocationsIds != null)
             {
-                return 0;
+                var duplicatedIdLines = duplicateLocationsIds.Select(issueLocation => issueLocation.LineNumber).JoinStr(",");
```

Comment:
I would place a space there for nice looking error message.
```suggestion
                var duplicatedIdLines = duplicateLocationsIds.Select(issueLocation => issueLocation.LineNumber).JoinStr(", ");
```